src/NSwag.SwaggerGeneration.AspNetCore/OperationResponseProcessor.cs

```diff
@@ -0,0 +1,116 @@
+// <copyright file="OperationResponseProcessor.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using NJsonSchema;
+using NJsonSchema.Infrastructure;
+using NSwag.SwaggerGeneration.Processors;
+using NSwag.SwaggerGeneration.Processors.Contexts;
+using NSwag.SwaggerGeneration.WebApi.Processors;
+
+namespace NSwag.SwaggerGeneration.AspNetCore
+{
+    /// <summary>Generates the operation's response objects based on reflection and the ResponseTypeAttribute, SwaggerResponseAttribute and ProducesResponseTypeAttribute attributes.</summary>
+    public class OperationResponseProcessor : IOperationProcessor
+    {
+        private readonly AspNetCoreToSwaggerGeneratorSettings _settings;
+
+        /// <summary>Initializes a new instance of the <see cref="OperationParameterProcessor"/> class.</summary>
+        /// <param name="settings">The settings.</param>
+        public OperationResponseProcessor(AspNetCoreToSwaggerGeneratorSettings settings)
+        {
+            _settings = settings;
+        }
+
+        /// <summary>Processes the specified method information.</summary>
+        /// <param name="operationProcessorContext"></param>
+        /// <returns>true if the operation should be added to the Swagger specification.</returns>
+        public async Task<bool> ProcessAsync(OperationProcessorContext operationProcessorContext)
+        {
+            if (!(operationProcessorContext is AspNetCoreOperationProcessorContext context))
+                return false;
+
+            var parameter = context.MethodInfo.ReturnParameter;
+            var successXmlDescription = await parameter.GetDescriptionAsync(parameter.GetCustomAttributes()).ConfigureAwait(false) ?? string.Empty;
+
+            var responseTypeAttributes = context.MethodInfo.GetCustomAttributes()
+                .Where(a => a.GetType().Name == "ResponseTypeAttribute" ||
+                            a.GetType().Name == "SwaggerResponseAttribute")
+                .Concat(context.MethodInfo.DeclaringType.GetTypeInfo().GetCustomAttributes()
+                    .Where(a => a.GetType().Name == "SwaggerResponseAttribute"))
+                .ToList();
+
+            var operation = context.OperationDescription.Operation;
+            foreach (var requestFormat in context.ApiDescription.SupportedRequestFormats)
+            {
+                if (!operation.Consumes.Contains(requestFormat.MediaType, StringComparer.OrdinalIgnoreCase))
+                {
+                    operation.Consumes.Add(requestFormat.MediaType);
+                }
+            }
+
+            if (responseTypeAttributes.Count > 0)
```

Comment:
@RSuter does this pivot seem right to you? If we do find any `SwaggerResponseAttributes` on an action, we'll exclusively use it in lieu of `SupportedResponseType` which is essentially `ConsumesAttribute`. The idea is to encourage you to use one of the two sets of attributes to represent the response format

Reply:
Looks good. Maybe we also need to add the SwaggerDefaulResponseAttribute to this list...